apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -1,47 +1,184 @@
-/*global BridgeServiceMixin */
+/*global bridge,
+         BridgeServiceMixin,
+         BroadcastChannel,
+         Draft,
+         Drafts,
+         performance,
+         streamClient
+*/
 'use strict';
 
-if (!('BridgeServiceMixin' in self)) {
-  importScripts('/services/js/bridge_service_mixin.js');
-}
+/**
+ * Type description for the ConversationSummary object that is return from the
+ * Conversation service.
+ * @typedef {Object} ConversationSummary
+ * @property {number} id Unique identifier of the the conversation.
+ * @property {string} body Short content of the last message in the
+ * conversation.
+ * @property {Array.<string>} participants List of the conversation
+ * participants.
+ * @property {Date} timestamp Conversation timestamp.
+ * @property {{ hasUnread: boolean, hasNewError: boolean }} status Conversation
+ * status that indicates whether conversation has unread messages or/and failed
+ * messages that user hasn't seen yet.
+ * @property {string} lastMessageType Type of the last message in the
+ * conversation. Can be either 'sms' or 'mms'.
+ * @property {Draft} draft If conversation is draft or has draft, then this
+ * field contains appropriate draft information, otherwise it's null.
+ */
+
+[
+  ['bridge', '/lib/bridge/bridge.js'],
+  ['streamClient', '/lib/bridge/plugins/stream/client.js'],
+  ['BridgeServiceMixin', '/services/js/bridge_service_mixin.js'],
+  ['Drafts', '/services/js/drafts.js']
+].forEach(([dependencyName, dependencyPath]) => {
+  if (!(dependencyName in self)) {
+    importScripts(dependencyPath);
+  }
+});
 
 (function(exports) {
-  const SERVICE_NAME = 'conversation-service';
+  const debug = 0 ?
+    (arg1, ...args) => console.log(`[ConversationService] ${arg1}`, ...args):
+    () => {};
 
-  const STREAMS = Object.freeze(
-    ['getAllConversations', 'getMessagesForConversation']
-  );
+  const mark = 0 ?
+    (markName) =>
+      performance.mark(`[ConversationService] ${markName}@sms.gaiamobile.org`):
+    () => {};
+
+  const priv = {
+    mobileMessageClients: Symbol('mobileMessageClients'),
+
+    getMobileMessageClient: Symbol('getMobileMessageClient')
+  };
+
+  const SERVICE_CONTRACT = Object.freeze({
+    name: 'conversation-service',
+
+    streams: Object.freeze(
+      ['getAllConversations', 'getMessagesForConversation']
+    ),
+
+    methods: Object.freeze([
+      'deleteConversations', 'deleteMessages', 'markConversationsAs',
+      'getConversationSummary', 'getMessage', 'findConversationFromAddress'
+    ]),
 
-  const METHODS = Object.freeze([
-    'deleteConversations', 'deleteMessages', 'markConversationsAs',
-    'getConversationSummary', 'getMessage', 'findConversationFromAddress'
-  ]);
+    events: Object.freeze(['message-change'])
+  });
 
-  const EVENTS = Object.freeze([
-    'message-change'
-  ]);
+  function draftToConversationSummary(draft) {
```

Comment:
> if we split threads.js, would that be easier for you to reuse ?

Ultimately yes, but I'm not sure if it's easy/good to do this in this PR, since I need two new fields ```draft``` and ```status``` that current master doesn't use.

> Not sure what you plan next, even we still need the data fetched/cached in the view, we might still need to convert thread handling logic into summary. So split these utils methods for reuse might help to avoid rewriting this part in Threads before we can enable service.

What I'm planning to do in the follow-ups is:
* Figure out reasonable setup for ConversationService where we see minimal, but acceptable perf regressions (host it window, maybe altogether with shim, maybe with custom endpoint);
* Switch Inbox to ConversationService. The main two changes here will be draft handling and ```unreadCount``` ---> ```status.hasUnread``` transition.
* DraftsService (once draft-dialog-removal patch lands).
* And somewhere here kill Drafts and InterInstanceEventDispatcher :)